### PR TITLE
Specify that we recomend typescript over JS for extensions

### PR DESCRIPTION
**Pull request by saulshanabrook**
While it's true that you can write extension in JS, most/all public ones are written in Typescript. So socially you really have to write them in typescript to feel at home. We should make that explicit and be honest about that.

### CODE REVIEW

File: docs/source/developer/extension_dev.rst [hunk 1]
```diff
@@ -21,6 +20,9 @@ A JupyterLab application is comprised of:
 -  A core Application object
 -  Plugins
 
+We recommend writing extensions in Typescript, but technically it doesn't matter what you write them in
+as long as they can be packaged as standard Javascript. 
+
```
Comment (jasongrout) [inline docs/source/developer/extension_dev.rst:25]: Good point. How about this language?
```suggestion
Extensions are distributed as JavaScript packages, so you can write extensions in JavaScript or any language that compiles to JavaScript. We recommend writing extensions in TypeScript, which is used for the JupyterLab core extensions and many popular community extensions.
```

Comment (saulshanabrook) [reply docs/source/developer/extension_dev.rst]: Much more eloquent, thank you Jason!

File: docs/source/developer/extension_dev.rst [hunk 2]
```diff
@@ -21,6 +20,7 @@ A JupyterLab application is comprised of:
 -  A core Application object
 -  Plugins
 
+Extensions are distributed as JavaScript packages, so you can write extensions in JavaScript or any language that compiles to JavaScript. We recommend writing extensions in TypeScript, which is used for the JupyterLab core extensions and many popular community extensions.
 Tutorials
```
Comment (jasongrout) [inline docs/source/developer/extension_dev.rst:24]: Oops, I accidentally deleted an important newline

```suggestion
Extensions are distributed as JavaScript packages, so you can write extensions in JavaScript or any language that compiles to JavaScript. We recommend writing extensions in TypeScript, which is used for the JupyterLab core extensions and many popular community extensions.

Tutorials
```

File: docs/source/developer/extension_dev.rst [hunk 3]
```diff
@@ -21,6 +20,8 @@ A JupyterLab application is comprised of:
 -  A core Application object
 -  Plugins
 
+Extensions are distributed as JavaScript packages, so you can write extensions in JavaScript or any language that compiles to JavaScript. We recommend writing extensions in TypeScript, which is used for the JupyterLab core extensions and many popular community extensions.
```
Comment (jasongrout) [inline docs/source/developer/extension_dev.rst:23]:
```suggestion
Extensions are distributed as JavaScript packages, so you can write extensions in JavaScript or any language that compiles to JavaScript. We recommend writing extensions in [TypeScript](https://www.typescriptlang.org/), which is used for the JupyterLab core extensions and many popular community extensions.
```

File: docs/source/developer/extension_dev.rst [hunk 4]
```diff
@@ -21,6 +20,8 @@ A JupyterLab application is comprised of:
 -  A core Application object
 -  Plugins
 
+Extensions are distributed as JavaScript packages, so you can write extensions in JavaScript or any language that compiles to JavaScript. We recommend writing extensions in [TypeScript](https://www.typescriptlang.org/), which is used for the JupyterLab core extensions and many popular community extensions.
```
Comment (jasongrout) [inline docs/source/developer/extension_dev.rst:23]:
```suggestion
Extensions are distributed as JavaScript packages, so you can write extensions in JavaScript or any language that compiles to JavaScript. We recommend writing extensions in `TypeScript <https://www.typescriptlang.org/>`_, which is used for the JupyterLab core extensions and many popular community extensions.
```

**Comment by blink1073** [comment]
I'm addressing the CI failures in #8433.

**Pull request merged**